### PR TITLE
Validate the `transform` method

### LINKED ISSUE #479: User validation for `transform`
### Problem Description

When a user uses `transform`, we should validate their input & throw appropriate warnings/errors.

### Expected behavior

Perform the following 3 validation checks:

#### 1. Transforming data before fitting it

Note: This also applies if the HyperTransformer was previously fit but the config has changed since then.

```python
ht = HyperTransformer()
ht.detect_initial_config(data)
ht.transform(data)
Error: The HyperTransformer is not ready to use. Please fit your data first using 'fit' or 'fit_transform'.
```

#### 2. Transforming the data before setting a config
```python
ht = HyperTransformer()
ht.transform(data)
Error: No config detected. Set the config using 'set_config' or pre-populate it automatically from your data using 'detect_initial_config' prior to fitting your data.
```

#### 3. The data does not have the exact same column names as the config (& fit data)
See #474 

```python
ht = HyperTransformer()
ht.detect_initial_config(data)
ht.fit(data)
ht.transform(different_data)
Error: The data you are trying to transform has different columns than the original data. Column names and their sdtypes
must be the same. Use the method 'get_config()' to see the expected values. 
```

### PR DESCRIPTION
**Pull request by fealho**
Resolve #479.

### CODE REVIEW

File: rdt/hyper_transformer.py [hunk 1]
```diff
@@ -565,17 +570,19 @@ def _validate_correctly_fitted(self, data):
...
+            )
 
         unknown_columns = self._subset(data.columns, self._input_columns, not_in=True)
         is_subset = any(column not in data.columns for column in self._input_columns)
         if unknown_columns or is_subset:
-            unknown_text = f' (unknown columns: {unknown_columns})' if unknown_columns else ''
             raise NotFittedError(
-                'The data you are trying to transform has different columns than the original '
-                f'fitted data{unknown_text}. Column names and their sdtypes must be the same. '
-                "Use the method 'get_config()' to see the expected values."
```
Comment (amontanez24) [inline rdt/hyper_transformer.py:578]: This line should still be included according to the issue

**Review by amontanez24 — CHANGES_REQUESTED** [review]
One small text change, but once that's done it should be ready to go!

**Review by amontanez24 — APPROVED** [review]
LGTM!

**Review by pvk-developer — APPROVED** [review]
:shipit:

**Pull request merged**